### PR TITLE
[E2E] NPM ECS environment

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

Implement E2E tests for ECS environment

### Additional Notes

need https://github.com/DataDog/test-infra-definitions/pull/626

### CODE REVIEW

File: test/new-e2e/pkg/environments/aws/ecs/ecs.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+			Type:      ssm.ParameterTypeSecureString,
+			Overwrite: pulumi.Bool(true),
+			Value:     awsEnv.AgentAPIKey(),
+		}, awsEnv.WithProviders(config.ProviderAWS))
+		if err != nil {
+			return err
+		}
+
+		// Deploy EC2 Agent
+		if linuxNodeGroupPresent {
```
Comment (hmahmood) [inline test/new-e2e/pkg/environments/aws/ecs/ecs.go:265]: What happens if this is `false`? Do we not have an agent running for non-Linux?

File: test/new-e2e/pkg/environments/aws/ecs/ecs.go [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+// WithoutAgent deactivates the creation of the Docker Agent
+func WithoutAgent() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = nil
+		return nil
+	}
+}
+
+// Run deploys a docker environment given a pulumi.Context
```
Comment (hmahmood) [inline test/new-e2e/pkg/environments/aws/ecs/ecs.go:160]: Does this comment need an update?

Comment (nplanel) [reply test/new-e2e/pkg/environments/aws/ecs/ecs.go:160]: thanks

File: test/new-e2e/pkg/environments/aws/ecs/ecs.go (hunk 1 above)
Comment (nplanel) [reply test/new-e2e/pkg/environments/aws/ecs/ecs.go:265]: for the moment the agent is not deployed on ecsWindowsNodeGroup but this where the else code will go

File: test/new-e2e/pkg/environments/aws/ecs/ecs.go [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+		infraShouldDeployFakeintakeWithLB: false,
+	}
+}
+
+// GetProvisionerParams return ProvisionerParams from options opts setup
+func GetProvisionerParams(opts ...ProvisionerOption) *ProvisionerParams {
+	params := newProvisionerParams()
+	err := optional.ApplyOptions(params, opts)
+	if err != nil {
+		panic(fmt.Errorf("unable to apply ProvisionerOption, err: %w", err))
```
[the remark this answers is not in the document]
Comment (nplanel) [reply test/new-e2e/pkg/environments/aws/ecs/ecs.go:64]: if we going there something is wrong in our code, so I think panic is appropriate, this is like that for other environments

[the remark this answers is not in the document]
Comment (pducolin) [reply test/new-e2e/pkg/environments/aws/ecs/ecs.go:64]: Yep we panic as we don't return an error when we call a provisioner. The provisioner handler then properly catch the panic and recovers.

File: test/new-e2e/pkg/environments/aws/ecs/ecs.go [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+	}
+	// Create cluster
+	ecsCluster, err := ecs.CreateEcsCluster(awsEnv, params.name)
+	if err != nil {
+		return err
+	}
+
+	// Export cluster’s properties
+	ctx.Export("ecs-cluster-name", ecsCluster.Name)
+	ctx.Export("ecs-cluster-arn", ecsCluster.Arn)
```
Comment (pducolin) [inline test/new-e2e/pkg/environments/aws/ecs/ecs.go:180]: **💬 suggestion**
Exported properties could be part of the environment. Here these are exported as not typed, but the provisioner is typed which could be confusing. I can take care of adding the type.

**Review by pducolin — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Pull request merged**